src/rdysetup.c

```diff
@@ -3,6 +3,44 @@
 #include <private/rdymemoryimpl.h>
 #include <rdycore.h>
 
+static PetscReal ConvertTimeToSeconds(PetscReal time, RDyTimeUnit time_unit) {
+  PetscFunctionBegin;
+
+  PetscReal time_in_sec;
+  PetscReal sec_in_min = 60.0;
+  PetscReal min_in_hr  = 60.0;
+  PetscReal hr_in_day  = 24.0;
+  PetscReal day_in_mon = 30.0;
+  PetscReal mon_in_yr  = 12.0;
+
+  switch (time_unit) {
+    case TIME_SECONDS:
+      printf("TIME_SECONDS\n");
+      time_in_sec = time;
+      break;
+    case TIME_MINUTES:
+      time_in_sec = time * sec_in_min;
+      break;
+    case TIME_HOURS:
+      time_in_sec = time * min_in_hr * sec_in_min;
+      break;
+    case TIME_DAYS:
+      time_in_sec = time * hr_in_day * min_in_hr * sec_in_min;
+      break;
+    case TIME_MONTHS:
+      time_in_sec = time * day_in_mon * hr_in_day * min_in_hr * sec_in_min;
+      break;
+    case TIME_YEARS:
+      time_in_sec = time * mon_in_yr * day_in_mon * hr_in_day * min_in_hr * sec_in_min;
+      break;
+    default:
+      PetscCall(PETSC_TRUE, PETSC_COMM_WORLD, PETSC_ERR_USER, "Unsupported time unit");
```

Comment:
Should this be `PetscCheck(PETSC_FALSE, PETSC_COMM_WORLD, ...)`?

Reply:
Just use `SETERRQ`